src/histolab/slide.py

```diff
@@ -245,142 +251,135 @@ def save_thumbnails(self, scale_factor=32, n=0):
 
     @property
-    def _avg_height_slide(self):
-        return sum(d["height"] for d in self.slides_dimensions) / self.total_slides
+    def _avg_size_slide(self) -> float:
+        return sum(d["size"] for d in self._slides_dimensions) / self.total_slides
 
     @property
-    def _avg_size_slide(self):
-        return sum(d["size"] for d in self.slides_dimensions) / self.total_slides
+    def _dimensions_stats(self) -> dict:
+        return {
+            "no_of_slides": self.total_slides,
+            "max_width": self._max_width_slide,
+            "max_height": self._max_height_slide,
+            "max_size": self._max_size_slide,
+            "min_width": self._min_width_slide,
+            "min_height": self._min_height_slide,
+            "min_size": self._min_size_slide,
+            "avg_width": self._avg_width_slide,
+            "avg_height": self._avg_height_slide,
+            "avg_size": self._avg_size_slide,
+        }
 
     @property
-    def _max_height_slide(self):
-        max_height = max(self.slides_dimensions, key=lambda x: x["height"])
+    def _max_height_slide(self) -> dict:
+        max_height = max(self._slides_dimensions, key=lambda x: x["height"])
         return {"slide": max_height["wsi"], "height": max_height["height"]}
 
     @property
-    def _max_size_slide(self):
-        max_size = max(self.slides_dimensions, key=lambda x: x["size"])
+    def _max_size_slide(self) -> dict:
+        max_size = max(self._slides_dimensions, key=lambda x: x["size"])
         return {"slide": max_size["wsi"], "size": max_size["size"]}
 
     @property
-    def _max_width_slide(self):
-        max_width = max(self.slides_dimensions, key=lambda x: x["width"])
+    def _max_width_slide(self) -> dict:
+        max_width = max(self._slides_dimensions, key=lambda x: x["width"])
         return {"slide": max_width["wsi"], "width": max_width["width"]}
 
     @property
-    def _min_width_slide(self):
-        min_width = min(self.slides_dimensions, key=lambda x: x["width"])
+    def _min_width_slide(self) -> dict:
+        min_width = min(self._slides_dimensions, key=lambda x: x["width"])
         return {"slide": min_width["wsi"], "width": min_width["width"]}
 
     @property
-    def _min_height_slide(self):
-        min_height = min(self.slides_dimensions, key=lambda x: x["height"])
+    def _min_height_slide(self) -> dict:
+        min_height = min(self._slides_dimensions, key=lambda x: x["height"])
         return {"slide": min_height["wsi"], "height": min_height["height"]}
 
     @property
-    def _min_size_slide(self):
-        min_size = min(self.slides_dimensions, key=lambda x: x["size"])
-        return {"slide": min_size["wsi"], "height": min_size["size"]}
+    def _min_size_slide(self) -> dict:
+        min_size = min(self._slides_dimensions, key=lambda x: x["size"])
+        return {"slide": min_size["wsi"], "size": min_size["size"]}
+
+    @property
+    def _slides_dimensions(self) -> List[dict]:
```

Comment:
It's not clear to me why we should have both `_wsi_dimensions_list ` and `_slides_dimensions` properties. They represent the same thing (if I'm not missing anything). I would keep `_slides_dimensions` (more structured).

Moreover, we should decide between "slide" or "wsi" naming. They represent the same thing but here it seems not

Reply:
I'm not sure that this is the best choice but slides_dimensions is used for the stats, but IMHO it's more easy to use to get avg, max etc...
Furthermore wsi_dimensions_list it's just a list of tuple of ints and so more fast than accessing to a dict key each time you need that values. So for now I would keep it separately til we'll find a more pythonic implementation of both

AGREE ON NAMING